Assignment1/Assignment1_QNo6/SolAssignmentNo1_QNo6.rb

```diff
@@ -0,0 +1,32 @@
+# => 6. Find out how to convert the string 'The Ruby Book is based on Ruby'
+# => to 'The Rails Book is based on Rails' ?
+# => author : Hrisikesh Singha Chowdhury
+
+# => definition of custom class
+class CustomClass
+  # => Fn to implement buisness logic
+  def replace_word
+    # => Asking users to give correct input strings
+    puts "Enter the sentence"
+    sent = gets.chomp
+    puts "Enter the old word to be replaced"
+    old_word = gets.chomp
+    puts "Enter the new word by which to be replaced"
+    new_word = gets.chomp
+    new_sent = sent.gsub!(old_word, new_word)
```

Comment:
@HrisikeshSinghaChowdhury If you use gsub! it will update the original string itself you then no need to store the return value in other string variable. you can directly use original string as it was updated.

Reply:
Please check the difference between gsub & gsub!

Reply:
here i want to check the case when the old word to be replaced is found or not.So i am storing the variable and later checking if it was nil as if nil the word is not found

Reply:
@HrisikeshSinghaChowdhury gsub & gsub! both will return string value as a return type so there is no point of calling `nil?` to check if nil

Reply:
i did not get this point.